### DISCUSSION TITLE
How can we change the database field synchronization `schema` more flexibly without losing data?

### DISCUSSION THREAD

**Discussion by evanlong-me** [prisma/prisma#20526] [Q&A] [answered]
### Question

You all know that every time you add or modify a field, you get the following prompt:

```base
Are you sure you want to reset your database? All data will be lost > (y/N)
```

### Expected behavior (optional)

I wish we had more flexibility to modify or add fields without affecting the data, and to synchronize the schema to get the TS type.

**Comment by ludralph** [comment, marked as answer]
Hi @evanlong0926 👋 

When you add or modify a field in your Prisma schema and then run `prisma migrate dev`, Prisma detects the schema drift and prompts you to reset your database. This prompt is there to ensure that you are aware of potential data loss that might occur due to the schema changes.

By default, when you add a non-nullable field or modify an existing field in a way that could lead to data loss, Prisma wants to make sure that you are conscious of the impact on the existing data. Therefore, it asks for confirmation before applying the migration.

To address your concerns about losing data and to provide more flexibility in modifying or adding fields, you can use the [Expand and Contract pattern](https://www.prisma.io/dataguide/types/relational/expand-and-contract-pattern) to evolve the schema without downtime. You can also see an example in the [documentation](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations#example-use-the-expand-and-contract-pattern-to-evolve-the-schema-without-downtime)

**Reply by evanlong-me** [reply]
Thank you very much. It's very useful for me.
